xptifw/samples/sycl_perf_collector/sycl_perf_collector.cpp

```diff
@@ -220,50 +221,91 @@ XPTI_CALLBACK_API void xptiTraceInit(unsigned int major_version,
+      else
+        ShowVerboseOutput = true;
+    } else {
+      ShowVerboseOutput = false;
+    }
+
+    const char *DebugFlag = std::getenv("XPTI_DEBUG");
+    // Check the environment variable for debug output and set the appropriate
+    // flag, if the variable is set
+    if (DebugFlag) {
+      if (std::stoi(DebugFlag) == 0)
+        ShowDebugInformation = false;
+      else {
+        ShowDebugInformation = true;
+        ShowVerboseOutput = true;
+      }
+    } else {
+      ShowDebugInformation = false;
+    }
+
     const char *ProfOutFile = std::getenv("XPTI_SYCL_PERF_OUTPUT");
-    if (ProfOutFile)
+    if (ShowVerboseOutput && ProfOutFile)
       std::cout << "XPTI_SYCL_PERF_OUTPUT=" << ProfOutFile << "\n";
     const char *StreamsToMonitor = std::getenv("XPTI_STREAMS");
-    if (StreamsToMonitor)
+    if (ShowVerboseOutput && StreamsToMonitor)
       std::cout << "XPTI_STREAMS=" << StreamsToMonitor << "\n";
     const char *ColorOutput = std::getenv("XPTI_STDOUT_USE_COLOR");
-    if (ColorOutput)
+    if (ShowVerboseOutput && ColorOutput)
       std::cout << "XPTI_STDOUT_USE_COLOR=" << ColorOutput << "\n";
     const char *FirstCallsToIgnore = std::getenv("XPTI_IGNORE_LIST");
-    if (FirstCallsToIgnore)
+    if (ShowVerboseOutput && FirstCallsToIgnore)
       std::cout << "XPTI_IGNORE_LIST=" << FirstCallsToIgnore << "\n";
     const char *SimulationOverheads = std::getenv("XPTI_SIMULATION");
-    if (SimulationOverheads)
+    if (ShowVerboseOutput && SimulationOverheads)
       std::cout << "XPTI_SIMULATION=" << SimulationOverheads << "\n";
     const char *CalibrationFlag = std::getenv("XPTI_CALIBRATE");
-    if (CalibrationFlag)
+    if (ShowVerboseOutput && CalibrationFlag)
       std::cout << "XPTI_SIMULATION=" << CalibrationFlag << "\n";
-    const char *VerboseFlag = std::getenv("XPTI_VERBOSE");
-    if (VerboseFlag)
+    if (ShowVerboseOutput && VerboseFlag)
       std::cout << "XPTI_VERBOSE=" << VerboseFlag << "\n";
+    if (ShowVerboseOutput && DebugFlag)
+      std::cout << "XPTI_DEBUG=" << DebugFlag << "\n";
+
+    // Check the environment variable for colored output and set the appropriate
+    // flag, if the variable is set
+    if (CalibrationFlag) {
+      if (std::stoi(CalibrationFlag) == 0)
+        CalibrationRun = false;
+      else
+        CalibrationRun = true;
+    } else {
+      CalibrationRun = false;
+    }
 
     // Get the streams to monitor from the environment variable
     // In order to determine if the environment variable contains valid stream
     // names, a catalog of all streams must be check against to validate the
     // strings
-    if (StreamsToMonitor) {
+    if (ShowVerboseOutput && StreamsToMonitor) {
       std::cout << "Monitoring streams: ";
 
       auto streams = D.decode(StreamsToMonitor);
       for (auto &s : streams) {
         std::cout << s << ", ";
         GAllStreams.add(s.c_str());
       }
-      std::cout << "null\n";
+      if (ShowVerboseOutput)
```

Comment:
I would suggest to wrap this prints to the method that will handle all required checks and conditions

Reply:
Eventually it can support DebugLevels = 0,1,2,3 etc. For now, this is sufficient. Need an infrastructure to start cost analysis.